crates/uv-distribution-types/src/prioritized_distribution.rs

```diff
@@ -623,6 +646,11 @@ impl WheelCompatibility {
         matches!(self, Self::Compatible(_, _, _))
     }
 
+    /// Return `true` if the distribution is excluded.
+    pub fn is_excluded(&self) -> bool {
+        matches!(self, Self::Incompatible(IncompatibleWheel::ExcludeNewer(_)))
+    }
```

Comment:
Is there a reason to not consider all `Incompatible()` instances "excluded"?

Reply:
Yeah...`ExcludeNewer` is kind of the odd one out here. "Incompatible" can also include things like "not compatible with the current platform". But for the `ExcludeNewer` distributions, we basically want to act like they don't exist at all. It would be easier if we could filter them out in the first place, but the problem is, we _do_ want to show dedicated error messages for them, so we have to propagate them throughout the system.